woocommerce/woocommerce-android
# Product Creation with AI: Functionality for Product Name Generation

Closes: #9806


### Description


This PR continues the https://github.com/woocommerce/woocommerce-android/pull/9802 and it's adding the functionality related to product name generation with AI.


### Testing instructions


1. Go to product list, create a new product, select "Create a product with AI",
2. Make sure the Product name step is shown, 
3. Tap "Suggest a name", make sure AI product name generation screen is shown,
4. Test the keyword entry: make sure AI button is disabled if keyword is empty
5. Enter some keyword then tap "Write it for me",
6. Make sure a generated product name is shown,
7. Test the Copy button,
8. Test the Regenerate button,
9. Test the Apply button, which should bring the generated product name into the product name field on the Product name step.


### Video


https://github.com/woocommerce/woocommerce-android/assets/266376/c72e2bce-cf4d-4261-be41-ec710f34f6a3
